src/formio/templates/fieldset.ejs

```diff
@@ -1,7 +1,7 @@
-<fieldset class="{{ctx.ofPrefix}}fieldset">
-  {% if (ctx.component.legend) { %}
+<fieldset class="{{ctx.ofPrefix}}fieldset {% if (!ctx.component.showHeader) { %}{{ctx.ofPrefix}}fieldset--no-header{% } %}">
+  {% if (ctx.component.label && ctx.component.showHeader) { %}
```

Comment:
as discussed to prevent breaking existing form configs, invert with `hideHeader` instead of `showHeader`